solidity/dashboard/src/services/token-staking.service.js

```diff
@@ -14,62 +14,75 @@ import { isSameEthAddress } from "../utils/general.utils"
 import { isEmptyArray } from "../utils/array.utils"
 import { getEventsFromTransaction } from "../utils/ethereum.utils"
 
-const fetchDelegatedTokensData = async () => {
-  const contractsLoaded = await ContractsLoaded
-  const { grantContract, stakingContract } = contractsLoaded
-  const web3 = await Web3Loaded
-  const { eth } = web3
-  const { defaultAccount: yourAddress } = eth
-
+const delegationInfoFromStakedEvents = async (address) => {
+  const { stakingContract } = await ContractsLoaded
   let operatorStakedEvents
   try {
     operatorStakedEvents = await stakingContract.getPastEvents(
       "OperatorStaked",
       {
         fromBlock: CONTRACT_DEPLOY_BLOCK_NUMBER.stakingContract,
-        filter: { operator: yourAddress },
+        filter: { operator: address },
       }
     )
   } catch (err) {}
 
-  let [stakedBalance, ownerAddress, initializationPeriod] = await Promise.all([
+  if (operatorStakedEvents.length === 0)
+    throw new Error("No OperatorStaked events found for address " + address)
+
+  const {
+    transactionHash: stakingTransactionHash,
+    returnValues: {
+      beneficiary: beneficiaryAddress,
+      authorizer: authorizerAddress,
+    },
+  } = operatorStakedEvents[0]
+
+  return {
+    stakingTransactionHash,
+    beneficiaryAddress,
+    authorizerAddress,
+  }
+}
+
+const fetchDelegatedTokensData = async () => {
+  const { grantContract, stakingContract } = await ContractsLoaded
+  const web3 = await Web3Loaded
+  const {
+    eth,
+    eth: { defaultAccount: yourAddress },
```

Comment:
Hmm, most places your address is derived from the page state, rather than from the web3 state. No reason to block right now, but particularly as we look to implement #2024, we're going to be leaning heavily on reading the “current address” from the page rather than the provider.